seismicpro/src/weathering_velocity/weathering_velocity.py

```diff
@@ -0,0 +1,598 @@
+        if negative_init:
+            raise ValueError(f"Init parameters contain negative values {negative_init}.")
+        negative_bounds = {key: val for key, val in bounds.items() if min(val) < 0}
+        if negative_bounds:
+            raise ValueError(f"Bounds parameters contain negative values {negative_bounds}.")
+        reversed_bounds = {key: [left, right] for key, [left, right] in bounds.items() if left > right}
+        if reversed_bounds:
+            raise ValueError(f"Left bound is greater than right bound for {reversed_bounds}.")
+        both_keys = {*init.keys()} & {*bounds.keys()}
+        outbounds_keys = {key for key in both_keys if init[key] < bounds[key][0] or init[key] > bounds[key][1]}
+        if outbounds_keys:
+            raise ValueError(f"Init parameters are out of the bounds for {outbounds_keys} key(s).")
+
+    def _validate_keys(self, checked_dict):
+        """Check the keys of given dict for a minimum quantity, an excessive, and an insufficient."""
+        expected_layers = len(checked_dict) // 2
+        if expected_layers < 1:
+            raise ValueError("Insufficient parameters to fit a weathering velocity curve.")
+        missing_keys = set(self._get_valid_keys(expected_layers)) - set(checked_dict.keys())
+        if missing_keys:
+            raise ValueError("Insufficient parameters to fit a weathering velocity curve. ",
+                            f"Check {missing_keys} key(s) or define `n_layers`")
+        excessive_keys = set(checked_dict.keys()) - set(self._get_valid_keys(expected_layers))
+        if excessive_keys:
+            raise ValueError(f"Excessive parameters to fit a weathering velocity curve. Remove {excessive_keys}.")
+
+    def _params_postprocceissing(self, params, ascending_velocity):
+        """Check the parameters and fix it if constraints are not met."""
+        params[self.n_layers:] *= 1000
+        # used `self._piecewise_offsets - params` because `self._piecewise_offsets` is the same array with max_offset
+        mask_offsets = self._piecewise_offsets[2:] - params[1:self.n_layers] < 0
+        params[1:self.n_layers][mask_offsets] = self._piecewise_offsets[2:][mask_offsets]
+        if ascending_velocity:
+            mask_velocity = np.diff(params[self.n_layers:]) < 0
+            params[self.n_layers:-1][mask_velocity] = params[self.n_layers + 1:][mask_velocity]
+        return params
+
+    def _ms_to_kms(self, params, as_array=True):
+        """Convert the velocity in the given valid dict of parameters from m/s to km/s."""
+        values = np.array(list(params.values()), dtype=float)
+        values[self.n_layers:] = values[self.n_layers:] / 1000
+        if as_array:
+            return values
+        return dict(zip(self._valid_keys, values))
+
+    @plotter(figsize=(10, 5))
+    def plot(self, *, ax=None, title=None, x_ticker=None, y_ticker=None, show_params=True, threshold_times=None,
+            compared_params=None, txt_kwargs=None, **kwargs):
```

Comment:
```suggestion
             compared_params=None, text_kwargs=None, **kwargs):
```
`text_kwargs` are not described in the method docs